python/pfs/pipe2d/generateCommands.py

```diff
@@ -0,0 +1,1493 @@
+                f"-j{processes}"]
+
+            command += getDevelopmentOptions() if devel else []
+            command += g.flatSource.getCommandLine(key="flatId")
+            command += g.arcSource.getCommandLine(key="arcId")
+            command += g.config.getCommandLine()
+
+            print(f"{shellCommand(command)}", file=fout)
+
+    def ingest(
+            self, logger: lsst.log.Log, fout: TextIO, dataDir: str, calib: str, rerun: str, copyMode: str):
+        """Put to ``fout`` commands to ingest this calib.
+
+        Parameters
+        ----------
+        logger : `lsst.log.Log`
+            Logger.
+        fout : `TextIO`
+            Output file where to write the command.
+        dataDir : `str`
+            Root of data repository.
+        calib : `str`
+            Name of output calibration directory.
+        rerun : `str`
+            Name of rerun.
+        copyMode : `str`
+            How to move files into calibration directory.
+        """
+        ingestCalibs(
+            logger, fout, "detectorMap", dataDir, calib, rerun, copyMode, self.validity,
+            overwrite=True)
+
+
+class _FiberTraceNoCombineSource(
+        CalibSource,
+        typeName="_fiberTraceNoCombine", commandName="constructFiberTrace.py"):
+    """Do not use this class explicitly.
+
+    This class is used internally by FiberTraceSource
+    to call constructFiberTrace.py for each group of input FITS files.
+    """
+    pass
+
+
+@export
+class FiberTraceSource(
+        CalibSource,
+        typeName="fiberTrace", commandName="combineFiberTraces.py"):
+    """Sources to construct fiberTrace.
+
+    How to construct fiberTrace is quite different
+    from how to construct other calibs.
+    YAML structure for fiberTrace is quite peculiar, accordingly.
+    See the docstring of ``fromYaml()`` for details.
+
+    Parameters
+    ----------
+    config : `CommandConfig`
+        Configurations used in constructing this calib.
+    groups : `Iterable[_FiberTraceNoCombineSource]`
+        Groups of sources, each group for non-combined fiberTraces.
+    validity : `int`
+        Valid days of the resulting calib.
+    """
+
+    __slots__ = ["groups"]
+
+    def __init__(self, config: CommandConfig, groups: Iterable[_FiberTraceNoCombineSource], validity: int):
+        super().__init__(config, SourceFilter(), validity)
+        self.groups = list(groups)
+
+    @classmethod
+    def fromYaml(cls, yamlBlock: Mapping[str, Any]) -> "FiberTraceSource":
+        """Construct ``FiberTraceSource`` from a YAML block.
+
+        Parameters
+        ----------
+        yamlBlock : `Mapping[str, Any]`
+            A block of a YAML data structure:
+
+            ``"group"``
+                List of Mapping[str, Any]`.
+                fiber traces are created for each group, and then combined.
+                Contents of each group are:
+
+                ``"id"``
+                    A string like ``"field=BIAS"`` or a list of such strings.
```

Comment:
Can I have something like:

    id:
        - "field=FLAT_ODD dateObs=2020-05-29"
        - "field=FLAT_EVEN dateObs=2020-05-29"

?

Reply:
If such is the syntax, and if one writes `id: ["field=FLAT_ODD", "dateObs=2020-05-29"]`, then the translation result will be `--id field=FLAT_ODD` for one group and `--id dateObs=2020-05-29` for another. This is inconsistent with other calibTypes. If I, for consistency, forbid `List[str]` from being the type of `id` field of other calibTypes, `id` field will be inconsistent to `config` field. I don't want `config` field to be a single string like `config: "key=value key=value"` because such a string has to be parsed and split by spaces when `value` may contain spaces in theory.

Reply:
We need to be able to select data based on multiple fields in the registry, like `--id field=FLAT_ODD dateObs=2020-05-29".

Reply:
The current design has this syntax:
```
group:
    -
        id: ["field=FLAT_ODD", "dateObs=2020-05-29"]
    -
        id: ["field=FLAT_EVEN", "dateObs=2020-05-29"]
```
Maybe I should revise the docstring.